specification/components/schemas/extensions/Extension-eRS-ReferralState.yaml

```diff
@@ -0,0 +1,97 @@
+title: Extension-eRS-ReferralState
+description: Extension to outline the current state of a ReferralRequest within e-RS.
+type: object
+required:
+  - url
+  - extension
+properties:
+  url:
+    type: string
+    enum:
+      - 'https://fhir.nhs.uk/STU3/StructureDefinition/Extension-eRS-ReferralState-1'
+    example: 'https://fhir.nhs.uk/STU3/StructureDefinition/Extension-eRS-ReferralState-1'
+  extension:
+    type: array
+    description: |
+      | Field                                      | Cardinality | Notes                                                                            |
+      | ------------------------------------------ | ----------- | -------------------------------------------------------------------------------- |
+      | `state`                                    | `1..1`      | Outlines the current state of the `ReferralRequest`                              |
+      | `reason`                                   | `0..1`      | Details the reason why the `ReferralRequest` is in this state, where appropriate |
+      | `comments`                                 | `0..1`      | Any comments around the current `ReferralRequest` state                          |
+    items:
+      anyOf:
+        - type: object
+          required:
+            - url
+            - valueCoding
+          properties:
+            url:
+              type: string
+              enum:
+                - state
+              example: 'state'
+            valueCoding:
+              type: object
+              required:
+                - system
+                - code
+                - display
+              properties:
+                system:
+                  type: string
+                  enum:
+                    - https://fhir.nhs.uk/STU3/CodeSystem/eRS-ReferralState-1
+                  example: https://fhir.nhs.uk/STU3/CodeSystem/eRS-ReferralState-1
+                code:
+                  type: string
+                  enum:
+                    - AWAITING_TRIAGE
+                    - BOOKED
+                    - UNBOOKED
+                  example: 'AWAITING_TRIAGE'
+                display:
+                  type: string
+                  enum:
+                    - 'Awaiting Triage'
+                    - 'Booked'
+                    - 'Unbooked'
+                  example: 'Awaiting Triage'
+        - type: object
+          required:
+            - url
+            - valueCoding
+          properties:
+            url:
+              type: string
+              enum:
+                - reason
+              example: 'reason'
+            valueCoding:
+              type: object
+              required:
+                - system
+                - code
+              properties:
+                system:
+                  type: string
+                  enum:
+                    - __baseurl__/FHIR/STU3/CodeSystem/APPOINTMENT-CANCELLATION-REASON
```

Comment:
It can be either a request cancellation reason or an appointment cancellation reason dependent on the defining state. The `reason` is the reason why the referral is in the current state, which could be either an appointment or request cancellation reason.